lib/main.js

```diff
@@ -0,0 +1,244 @@
+    process.env[`INPUT_${name.replace(/ /g, '_').toUpperCase()}`] = value;
+
+  // setInput("appName", "amber-test-container")
+  setInput("appName", "error-test")
+  setInput("resourceGroup", "amber-test-container")
+  // setInput("imageID", "mcr.microsoft.com/azuredocs/containerapps-helloworld:latest")
+  setInput("imageID", "ambertest.azurecr.io/testapp:errors")
+  setInput("logAnalyticsWorkspace", "amber-test-app-logs")
+  setInput("revisionSuffix", getTime())
+  setInput("stepPct", "50")
+  setInput("stepTime", "5")
+  setInput("monitorInterval", "1")
+  setInput("canaryDeploy", "true")
+  setInput("finalPct", "100")
+  setInput("monitors", "https://www.google.com")
+  setInput("errorThreshold", "0")
+}
+setInputs() // for local runs only to set environment variables
+// END TEST CODE
+
+// grab input values from github environment
+const APP = core.getInput('appName') // REQD application name
+const RG = core.getInput('resourceGroup') // REQD resource group name
+const IMAGE = core.getInput('imageID') // REQD image to update with
+const REV_SUFFIX = core.getInput('revisionSuffix') // REQD new revision suffix
+const LOG_ANALYTICS = core.getInput('logAnalyticsWorkspace') // log analytic workspace app is in
+const CANARY = core.getBooleanInput('canaryDeploy') // whether to use canary deployment
+const STEP_PCT = parseInt(core.getInput('stepPct')) // step pct of traffic going to new revision
+const STEP_TIME = parseFloat(core.getInput('stepTime')) // step time between traffic shifts
+const FINAL_PCT = parseInt(core.getInput('finalPct')) // final pct of traffic going to new revision
+const MONITORS = core.getInput('monitors').replace(/ +/g, '').split(/,|;|\n/) // api endpoints to hit
+const MONITOR_INTERVAL = parseFloat(core.getInput('monitorInterval')) // time between monitoring
+const ERROR_THRESHOLD = parseFloat(core.getInput('errorThreshold')) // max error tolerance on http pings
+const TIME_CONVERSION = 60000 // conversion factor from minutes to milliseconds
+
+let RESOURCE = ""
+let prevTrafficSettings = ""
+
+// utility function for async exec command
+const execute = function (command) {
+  return new Promise((resolve, reject) => {
+    console.log(`Running command ${command} at ${getDateTime()}`)
+    exec(command, (error, stdout, stderr) => {
+      if (error) {
+        if (error.signal == 'SIGTERM') {
+          resolve('Process was killed');
+        } else {
+          reject(error);
+        }
+      } else {
+        core.debug(stdout)
+        core.debug(stderr)
+        resolve(stdout);
+      }
+    });
+  })
+}
+
+// sleep callback with delay in ms
+const sleep = (delay) => new Promise(r => setTimeout(r, delay));
+
+// get date and time for queries
+function getDateTime() {
+  var today = new Date();
+  var date = today.getFullYear() + '-' + (today.getMonth() + 1) + '-' + today.getDate();
+  var time = today.getHours() + ":" + today.getMinutes() + ":" + today.getSeconds();
+  return date + 'T' + time;
```

Comment:
Infact, you can get YYYY-MM-DDThh:mm:ss.sssZ format by simply doing 
today.toISOString()